text/0000-layer-origin-metadata.md

```diff
@@ -0,0 +1,225 @@
+# Meta
+[meta]: #meta
+- Name: Layer Origin Metadata
+- Start Date: 2020-07-09
+- Author(s): pclalv
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: https://github.com/buildpacks/rfcs/pull/63
+
+# Summary
+[summary]: #summary
+
+<!-- One paragraph explanation of the feature. -->
+
+Exposing layer origin metadata in `report.toml` will improve layer
+reproducibility and, in turn, build reproducibility.
+
+# Motivation
+[motivation]: #motivation
+
+- Layer generation can be influenced by the state of the file system,
+  namely by the state of the application directory, other layers, and
+  the stack's build image
+- Layers aren't reproducible, because we can't know enough about the
```

Comment:
I think this gets more complicated with things mounted by a platform like `kpack` or `pack`. Imagine the https://github.com/buildpacks/rfcs/pull/87's example of a cert buildpack. Does the platform have to document things about the cert volume it mounted? 

Similarly, what about `<platform>/env/` values? Those changing will change the resulting build as well 🤔

Reply:
Sorry, I think at one point we talked about the [cert provider](https://github.com/buildpacks/rfcs/blob/app-mixins/text/0000-app-mixins-stack-buildpacks.md#ca-cert-provider-buildpack) may do something like read from `/<platform>/certs`. If you imagine a platform where you upload certs for all your apps to use and that buildpack pulls them in, there maybe should be a representation of that in this report so you can tell something change. The same could be true for `/platform/secrets` mounts.

I like the immutable representation idea for `<platform>/env` (and others I suppose?).

Reply:
it'd be nice if the platform could reference `env`, `certs` and whatever else as layers. you put those platform auxiliary layers existed on a registry, and then you've also solved the problem of durability.

unfortunately, that seems pretty out-of-scope - the ideas in this RFC depend on those ideas being implemented first, because yeah, otherwise we've got mystery files floating around.